lb_content_resolver/content_resolver.py

```diff
@@ -58,33 +67,30 @@ def scan(self, music_dir):
         self.error = 0
         self.skipped = 0
 
-        self.open_index()
-        self.writer = self.ix.writer()
-
+        self.open_db()
         self.traverse("")
-
-        self.writer.commit()
-        self.writer = None
-        self.close_index()
+        self.close_db()
 
         print("Checked %s tracks:" % self.total)
         print("  %5d tracks not changed since last run" % self.not_changed)
         print("  %5d tracks added" % self.added)
         print("  %5d tracks updated" % self.updated)
         print("  %5d tracks could not be read" % self.error)
-        print("  %5d files are not supported" % self.skipped)
-        if self.total != self.not_changed + self.updated + self.added + self.error + self.skipped:
+        if self.total != self.not_changed + self.updated + self.added + self.error:
             print("And for some reason these numbers don't add up to the total number of tracks. Hmmm.")
 
     def traverse(self, relative_path):
+        """
+            This recursive function searches for audio files and descends into sub directories 
+        """
 
         if not relative_path:
             fullpath = self.music_dir
         else:
             fullpath = os.path.join(self.music_dir, relative_path)
 
         for f in os.listdir(fullpath):
-            if f in ['.', '..']:
+            if f in ['.', '..'] or f.lower().endswith("jpg"):
```

Comment:
For later but we could have a list of known common, non-audio file extensions and check them here. Or we go with the other approach and only consider files who's extensions match what is supported and ignore everything else.

Reply:
Agreed, later. Lets learn how this works and what other people think about it.